engine/executor/server.go

```diff
@@ -212,6 +215,12 @@ func (s *Server) makeTask(
 		log.Error("Failed to create worker", zap.Error(err))
 		return nil, err
 	}
+	if _, ok := newWorker.(framework.BaseJobMaster); ok {
+		err := precheckMasterMeta(dctx, globalRegistry, workerID, workerType)
```

Comment:
Seems this PR is to solve `CreateWorker` return a `nil` `err` but the error is written to metastore. How could this happen? the jobmaster is not initialized, no business logic can run and no error can be written to metastore.

Reply:
It solves the scenario that `Tick` or `InitImpl` of business logic returns error.

Reply:
CreateWorker will not call `Tick` or `InitImpl`? I noticed when code runs here, `OnOpenAPIInitialized` is not called, and from our doc `OnOpenAPIInitialized` is the first callback function.

Reply:
error in metadata is not persisted here, it **was recorded** by last round of job master.